jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -690,9 +690,11 @@ ecma_builtin_number_prototype_object_to_fixed (ecma_number_t this_num, /**< this
     return ecma_builtin_number_prototype_object_to_string (this_num, NULL, 0);
   }
 
+  const lit_utf8_size_t total_digit_count = (lit_utf8_size_t) ((int32_t) num_digits + exponent);
```

Comment:
I have a question: if exponent can be < 0, what ensures it num_digits < exponent. If not, why it is not casted to lit_utf8_size_t and do an unsigned addition? Also, why do we need a local variable for this?

Reply:
Exponent is always non-negative, patch has been updated.